elastalert/test_rule.py

```diff
@@ -403,7 +403,7 @@ def run_rule_test(self):
                 with open(args.save, 'wb') as data_file:
                     # Add _id to _source for dump
                     [doc['_source'].update({'_id': doc['_id']}) for doc in hits]
-                    data_file.write(json.dumps([doc['_source'] for doc in hits], indent='    '))
+                    data_file.write(json.dumps([doc['_source'] for doc in hits], indent=4))
             if args.use_downloaded:
```

Comment:
From:  https://github.com/python/cpython/blob/master/Lib/json/encoder.py#L289

```
            newline_indent = '\n' + _indent * _current_indent_level
```

Without this being a number,

```
Traceback (most recent call last):
  File "/usr/local/bin/elastalert-test-rule", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/elastalert/test_rule.py", line 421, in main
    test_instance.run_rule_test()
  File "/usr/local/lib/python2.7/site-packages/elastalert/test_rule.py", line 406, in run_rule_test
    data_file.write(json.dumps([doc['_source'] for doc in hits], indent='    '))
  File "/usr/local/lib/python2.7/json/__init__.py", line 251, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/local/lib/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
  File "/usr/local/lib/python2.7/json/encoder.py", line 431, in _iterencode
    for chunk in _iterencode_list(o, _current_indent_level):
  File "/usr/local/lib/python2.7/json/encoder.py", line 300, in _iterencode_list
    newline_indent = '\n' + (' ' * (_indent * _current_indent_level))
TypeError: can't multiply sequence by non-int of type 'str'
Exited with code 1
```